dsgrid/config/time_dimension_base_config.py

```diff
@@ -110,3 +123,114 @@ def list_expected_dataset_timestamps(self):
             List of tuples of columns representing time in the load_data table.
 
         """
+
+    def _convert_time_to_project_time_interval(
+        self, df=None, project_time_dim=None, wrap_time=True
+    ):
+        """Shift time to match project time based on TimeIntervalType
+        If time range spills over into another year after time interval alignment,
+        the time range will be wrapped around so it's bounded within the year
+        """
+        if project_time_dim is None:
+            return df
+
+        dtime_interval = self.get_time_interval_type()
+        ptime_interval = project_time_dim.get_time_interval_type()
+        time_col = project_time_dim.get_timestamp_load_data_columns()
+
+        assert len(time_col) == 1, time_col
+        time_col = time_col[0]
+
+        df = self._align_time_interval_type(
+            df, time_col, dtime_interval, ptime_interval, self.get_frequency()
+        )
+
+        if wrap_time:
+            df = self._apply_time_wrap(df, project_time_dim)
+
+        return df
+
+    @staticmethod
+    def _align_time_interval_type(
+        df, time_column, from_time_interval, to_time_interval, time_step, new_time_column=None
+    ):
+        """
+        Shift time_column by time_step in df as needed by comparing from_time_interval to to_time_interval.
+        If new_time_column is None, time_column is shifted in place, else shifted time is added as new_time_column in df
+        """
+        if from_time_interval == to_time_interval:
+            return df
+
+        if new_time_column is None:
+            new_time_column = time_column
+
+        match (from_time_interval, to_time_interval):
```

Comment:
What about INSTANTANEOUS?

Reply:
We do nothing I think

Reply:
This code certainly does nothing in that case. Is that correct? Should we log anything? If this is correct then I suggest adding a comment here indicating that.

Reply:
@elainethale The conclusion from Lixi and I is that we should raise an exception in the case of a mismatch with instantaneous. There is a related comment on this subject at the top of the PR. Are you OK with punting on this question?

Reply:
Yes, I am okay with punting. Punting can be a TODO, logging a warning, or raising an exception. Up to you all.

Reply:
I added an exception and opened #239